docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+// This field must be omitted if `ipam.mode` is `Disabled`.
+//
+// +optional
+Subnets DualStackCIDRs `json:"subnets,omitempty"`
+
++ // excludeSubnets specifies a list of CIDRs removed from the specified CIDRs in `subnets`.
++ // excludeSubnets is optional. When omitted no IP address is excluded and all IP address specified by `subnets` subject to be automatically assigned.
++ // It is still allowed to request IPs from this range through static IP assignment.
++ // Each item should be in range of the specified CIDR(s) in `subnets`.
++ // The maximal exceptions allowed is 25.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=25
++ ExcludeSubnets []CIDR `json:"excludeSubnets,omitempty"`
+
+// JoinSubnets are used inside the OVN network topology.
+//
+// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the platform will choose a reasonable default which is subject to change over time.
+//
+// +optional
+JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
+
++ // infrastructureSubnets specifies a list of internal CIDR ranges that OVN-Kubernetes will reserve for internal network infrastructure.
++ // Any IP addresses within these ranges cannot be assigned to workloads.
++ // When omitted, OVN-Kubernetes will automatically allocate IP addresses from `subnets` for its infrastructure needs.
++ // When `excludeSubnets` is also specified the CIDRs cannot overlap.
++ // When `defaultGatewayIPs` is also specified  the default gateway IPs cannot be a part of any of the CIDRs.
++ // Each item should be in range of the specified CIDR(s) in `subnets`.
++ // The maximal exceptions allowed is 10.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=10
++ InfrastructureSubnets []CIDR `json:"infrastructureSubnets,omitempty"`
+
++ // defaultGatewayIPs specifies the default gateway IP used in the internal OVN topology.
++ //
++ // Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
++ // This field is only allowed for "Primary" network.
++ // It is not recommended to set this field without explicit need and understanding of the OVN network topology.
++ // When omitted, the first IP address from the pod network subnet is used.
++ //
++ // +optional
++ DefaultGatewayIPs DualStackIPs `json:"defaultGatewayIPs,omitempty"`
```

Comment:
the MAC address will be derived from this custom gatewayIP we pick? can we call this aspect out.. user can't also request the MAC address for the gateway right? I hope its going to be conversion from the gatewayIP?

Reply:
> the MAC address will be derived from this custom gatewayIP we pick? can we call this aspect out.. user can't also request the MAC address for the gateway right? I hope its going to be conversion from the gatewayIP?

There is no change to how the GW MAC is generated, so same as today the MAC address of the gateway will be calculated from the IP address.